lib/debugger/hookWorld.luau

```diff
@@ -1,14 +1,37 @@
+--#selene:allow(empty_loop)
 local useCurrentSystem = require(script.Parent.Parent.topoRuntime).useCurrentSystem
 local World = require(script.Parent.Parent.World)
+local rollingAverage = require(script.Parent.Parent.rollingAverage)
 
 local originalQuery = World.query
 
 local function hookWorld(debugger)
 	World.query = function(world, ...)
 		if useCurrentSystem() == debugger.debugSystem then
+			local start = os.clock()
+
+			for _ in originalQuery(world, ...) do
+			end
+
+			local duration = os.clock() - start
+			local file, line = debug.info(2, "sl")
+
+			if debugger._queryDurationSamples[file .. line] == nil then
+				debugger._queryDurationSamples[file .. line] = {}
+			end
```

Comment:
Small nitpick: Might be better to just cache `debugger._queryDurationSamples[file .. line]`

Reply:
agreed, it should be this
```lua
local key = file..line
local samples = debugger._queryDurationSamples
local sample = samples[key]
if not sample then 
    sample = {}
    samples[key] = sample
end

rollingAverage.addSample(sample, duration)
```